### PR TITLE
Support ephemeral storage resource validation for tasks

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
Support ephemeral storage resource validation for tasks

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/494

## Follow-up issue
_NA_

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -16,7 +16,7 @@ require (
 	github.com/coreos/go-oidc v2.2.1+incompatible
 	github.com/dgrijalva/jwt-go v3.2.0+incompatible
 	github.com/evanphx/json-patch v4.9.0+incompatible
-	github.com/flyteorg/flyteidl v0.19.14
+	github.com/flyteorg/flyteidl v0.19.19-0.20210804191154-47e32aa6d059
```
Comment (katrogan) [inline go.mod:19]: won't merge until updating

File: pkg/manager/impl/execution_manager.go [hunk 2]
```diff
@@ -297,6 +305,19 @@ func assignResourcesIfUnset(ctx context.Context, identifier *core.Identifier,
 		logger.Debugf(ctx, "Setting 'memory' for [%+v] to %s", identifier, platformValues.Memory)
 		resourceEntries = append(resourceEntries, memoryResource)
 	}
+	if ephemeralStorageindex < 0 && platformValues.EphemeralStorage != "" {
```
[the remark this answers is not in the document]
Comment (katrogan) [reply pkg/manager/impl/execution_manager.go:308]: true, i suppose for platform defaults ephemeral storage doesn't need to be set. updated

File: pkg/manager/impl/execution_manager.go [hunk 3]
```diff
@@ -250,23 +251,30 @@ func createTaskDefaultLimits(ctx context.Context, task *core.CompiledTask,
 	if systemResourceLimits.GPU != "" {
 		taskResourceLimits.GPU = systemResourceLimits.GPU
 	}
+	if systemResourceLimits.EphemeralStorage != "" {
+		taskResourceLimits.EphemeralStorage = systemResourceLimits.EphemeralStorage
+	} else if ephemeralStorageIndex >= 0 {
```
[the remark this answers is not in the document]
Comment (katrogan) [reply pkg/manager/impl/execution_manager.go:256]: done

File: pkg/manager/impl/execution_manager.go [hunk 4]
```diff
@@ -250,23 +251,30 @@ func createTaskDefaultLimits(ctx context.Context, task *core.CompiledTask,
 	if systemResourceLimits.GPU != "" {
```
[the remark this answers is not in the document]
Comment (katrogan) [reply pkg/manager/impl/execution_manager.go]: good point, this seems wrong. updated to only assign if not assigned.

**Comment by katrogan** [comment]
PTAL @EngHabu

File: pkg/manager/impl/validation/task_validator.go [hunk 5]
```diff
@@ -158,6 +162,11 @@ func requestedResourcesToQuantity(
 				return nil, errors.NewFlyteAdminErrorf(codes.InvalidArgument,
 					"gpu for [%+v] must be a whole number, got: %s instead", identifier, limitEntry.Value)
 			}
+		case core.Resources_EPHEMERAL_STORAGE:
+			err := addResourceEntryToMap(identifier, limitEntry, &requestedToQuantity)
```
Comment (pmahindrakar-oss) [inline pkg/manager/impl/validation/task_validator.go:166]: Does the requestedToQuantity need to be passed as a pointer here. Do we need the function to modify the value

**Review by pmahindrakar-oss — APPROVED** [review]

File: pkg/manager/impl/validation/task_validator.go (hunk 5 above)
Comment (katrogan) [reply pkg/manager/impl/validation/task_validator.go:166]: yeah it's adding the entry to the map arg

**Pull request merged**